R/format_GLA.R

```diff
@@ -0,0 +1,688 @@
+  force(db)
+
+  #### Determine species and population codes for filtering
+  if(is.null(species)){
+
+    species_filter <- NULL
+
+  } else {
+
+    species_filter <- species
+
+  }
+
+  ## Only keeping 5 main populations, so these will be the default filter ("CAS", "GAR", "SAL","KEL", "SCE")
+  ## Otherwise, use the specified pop filter
+  if(is.null(pop)){
+
+    pop_filter <- c("CAS", "GAR", "SAL","KEL", "SCE")
+
+  } else {
+
+    pop_filter <- pop
+
+  }
+
+  start_time <- Sys.time()
+
+  ## Set options
+  options(dplyr.summarise.inform = FALSE)
+
+  ## Read experiment classification table
+  expID_tab <- utils::read.csv(file = paste0(db, "/GLA_experiment_groups.csv"))
+
+  ## Read in primary data from brood records
+  nest_data <- readxl::read_xlsx(path = paste0(db, "/GLA_PrimaryData_Nest.xlsx"), guess = 5000) %>%
+    janitor::clean_names(case = "upper_camel") %>%
+    janitor::remove_empty(which = "rows") %>%
+
+    ## Reformat and rename important variables
+    dplyr::mutate(BreedingSeason = as.integer(.data$Yr),
+                  LocationID = as.character(.data$NestboxNumber),
+                  Species = as.character(.data$Species),
+                  PopID = as.character(.data$Site)) %>%
+
+    ## Adjust dates
+    dplyr::rowwise() %>%
+
+    ## There are two formats for dates.
+    ## The first can be handled with Lubridate, the second needs to be rearranged
+    dplyr::mutate(FirstEggDate = suppressWarnings(dplyr::case_when(grepl("/", .data$FirstEggDate) ~ lubridate::dmy(.data$FirstEggDate, quiet = TRUE),
+                                                                   TRUE ~ lubridate::ymd(paste(unlist(stringr::str_split(as.character(janitor::excel_numeric_to_date(as.numeric(.data$FirstEggDate))), pattern = "-"))[c(1,3,2)], collapse = "-"), quiet = TRUE))),
+
+                  LayingComplete = suppressWarnings(dplyr::case_when(grepl("/", .data$LayingComplete) ~ lubridate::dmy(.data$LayingComplete, quiet = TRUE),
+                                                                     TRUE ~ lubridate::ymd(paste(unlist(stringr::str_split(as.character(janitor::excel_numeric_to_date(as.numeric(.data$LayingComplete))), pattern = "-"))[c(1,3,2)], collapse = "-"), quiet = TRUE))),
+
+                  ObservedHatch = suppressWarnings(dplyr::case_when(grepl("/", .data$ObservedHatch) ~ lubridate::dmy(.data$ObservedHatch, quiet = TRUE),
+                                                                    TRUE ~ lubridate::ymd(paste(unlist(stringr::str_split(as.character(janitor::excel_numeric_to_date(as.numeric(.data$ObservedHatch))), pattern = "-"))[c(1,3,2)], collapse = "-"), quiet = TRUE))),
+
+                  Hatchlings = as.integer(.data$Hatchlings),
+                  Fledglings = as.integer(.data$Fledglings),
+                  MaleRing = as.character(.data$MaleRing),
+                  FemaleRing = as.character(.data$FemaleRing),
+                  ClutchSize = as.integer(.data$ClutchSize)) %>%
```

Comment:
Does this mean they've accidentally stored some dates in US format (M/D/Y) and some in non-US format (D/M/Y)? What a nightmare! 😆

This could be a bit streamlined by using `mutate(across())`. Also, after the unusual date issue we can `ungroup()` because it is much slower to do things rowwise. See my suggestion below. 

```suggestion
    dplyr::rowwise() %>%
    
    ## There are two formats for dates.
    ## The first can be handled with Lubridate, the second needs to be rearranged
    dplyr::mutate(dplyr::across(.cols = c(.data$FirstEggDate,
                                   .data$LayingComplete,
                                   .data$ObservedHatch),
                         .fns = ~suppressWarnings(dplyr::case_when(grepl("/", .) ~ lubridate::dmy(., quiet = TRUE),
                                                                   TRUE ~ lubridate::ymd(paste(unlist(stringr::str_split(as.character(janitor::excel_numeric_to_date(as.numeric(.))), pattern = "-"))[c(1,3,2)], collapse = "-"), quiet = TRUE))))) %>%
    dplyr::ungroup() %>%
    dplyr::mutate(
      Hatchlings = as.integer(.data$Hatchlings),
      Fledglings = as.integer(.data$Fledglings),
      MaleRing = as.character(.data$MaleRing),
      FemaleRing = as.character(.data$FemaleRing),
      ClutchSize = as.integer(.data$ClutchSize)) %>% 
```

Reply:
Indeed. It is a pain and this chunk to the process the dates is the slowest part of the code. 

Your suggestions make it look much cleaner though.